Fix for x-ms-examples extractor

sergey-shandar opened the pull request:
#309

veronicagg commented on documentation/source-map-design.md:
```diff
@@ -1,59 +0,0 @@
-# Source Map Design
```
is deleting this file intentional?

veronicagg commented on lib/util/validationError.ts:
```diff
@@ -15,7 +15,7 @@ export class ValidationError {
    *
    * @param name Validation Error Name
-   * @param severity The
+   * @param severity The severity
```
severity of the error?

veronicagg commented on test/xMsExamplesExtractor/databox.json:
```diff
@@ -0,0 +1,2157 @@
+{
```
should the spec be simplified to target what we need to test? seems like a long spec.

veronicagg replied:
should we have a folder "databox"  or "simple" and put all the assets corresponding to it there? the swagger and session records inside?

sergey-shandar replied:
Since we had no tests for x-ms-examples-extractor, I would prefer to keep at least one set of files that gives us good coverage.

veronicagg commented on documentation/suppression-examples.md:
```diff
@@ -0,0 +1,38 @@
+# Suppression Examples
```
is adding this file as part of this PR intentional?

sergey-shandar replied:
I'm going to move the file to another branch.

veronicagg commented:
@sergey-shandar thanks, any comments on the question above? "what caused the issue reported by the customer? and how's this change fixing it? The error in the linked issue doesn't have much detail."

sergey-shandar commented:
@veronicagg the problem was caused by replacing `getEntries` by `entries` functions. There was already a varible with the same name `entries` which has type `any`. I've renamed the variable to `recordingEntries` and added export to the `entries` function.

https://github.com/Azure/oav/blob/a7c05f3e061db6ca3f74969013b1d4ab4f5baabd/lib/xMsExampleExtractor.ts#L126

In general, the cause of this problem is weak typing (using `any` type) and no test coverage.
